builder/evaluator.go

```diff
@@ -191,6 +197,18 @@ func (b *builder) Run(context io.Reader) (string, error) {
 		}
 	}
 
+	// check if there are any leftover build-args that were passed but not
+	// consumed during build. Return an error, if there are any.
+	leftoverArgs := []string{}
+	for arg := range b.buildArgs {
+		if !b.isBuildArgAllowed(arg) {
+			leftoverArgs = append(leftoverArgs, arg)
+		}
+	}
+	if len(leftoverArgs) > 0 {
+		return "", fmt.Errorf("One or more build-args were not consumed by the build: %v", leftoverArgs)
```

Comment:
We need to make it more clear that this is an error and command failed. Currently its bit like a warning because the image is created. But it is actually an error because the image was left untagged.